### PR TITLE
Serialization/Deserialization

### PR DESCRIPTION
**Pull request by tayoogunbiyi**
Aims to close https://github.com/astaxie/beego/issues/4088

### CODE REVIEW

File: pkg/orm/models_info_f.go [hunk 1]
```diff
@@ -241,7 +242,27 @@ checkType:
...
 
-		fieldType, err = getFieldType(addrField)
+		serializerKey := sf.Tag.Get(serializerStructTagName)
+
+		if serializerKey != "" {
+			tags["type"] = "text"
+			fieldType = TypeTextField
+			err = nil
+
+			if fi.auto || fi.pk {
```
Comment (flycash) [inline pkg/orm/models_info_f.go:252]: I think we don't need to check those. Users should be responsible for making sure that the serializer works for this field.

File: pkg/orm/serializers.go [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	"encoding/json"
+)
+
+// SerializerMap maps from the serializer's string representation to the Serializer
+var serializerMap = make(map[string]FieldSerializer)
+
+// FieldSerializer is the interface which all serializers must implement
+type FieldSerializer interface {
+	Serialize(obj interface{}) string
+	Deserialize(fieldContent string) interface{}
```
Comment (flycash) [inline pkg/orm/serializers.go:13]: If you want to deserialize, you may need the field's metadata.

**Comment by tayoogunbiyi** [comment]
Next step is to handle nil fields appropriately, as well as add more tests for the fields in the comment mentioned above.

**Comment by flycash** [comment]
are you still working on this? It's hard to verify without unit tests.

**Comment by tayoogunbiyi** [comment]
I've not been able to work on this for a while. Please feel free to re-assign

**Pull request closed without merging**